src/dataflow-types/src/types.rs

```diff
@@ -1307,42 +1313,115 @@ pub mod sources {
+        pub external_id: Option<String>,
         /// The AWS region to use.
-        pub region: String,
+        ///
+        /// Uses the default region (looking at env vars, config files, etc) if not provideed.
+        pub region: Option<String>,
         /// The custom AWS endpoint to use, if any.
         pub endpoint: Option<SerdeUri>,
     }
 
-    /// Static AWS credentials for a source or sink.
+    /// AWS credentials for a source or sink.
     #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
-    pub struct AwsCredentials {
-        pub access_key_id: String,
-        pub secret_access_key: String,
-        pub session_token: Option<String>,
+    pub enum AwsCredentials {
+        /// Look for credentials using the [default provider][docs] chain
+        ///
+        /// [docs]: https://docs.rs/aws-config/latest/aws_config/default_provider/index.html
+        Default,
+        /// Load credentials using the given named profile
+        Profile { profile_name: String },
+        /// Use the credentials provided as WITH options
+        Static {
+            access_key_id: String,
+            secret_access_key: String,
+            session_token: Option<String>,
+        },
     }
 
     impl AwsConfig {
         /// Loads the AWS SDK configuration object from the environment, then
         /// applies the overrides from this object.
         pub async fn load(&self) -> mz_aws_util::config::AwsConfig {
-            use aws_smithy_http::endpoint::Endpoint;
-            use aws_types::credentials::SharedCredentialsProvider;
-            use aws_types::region::Region;
-
-            let mut loader = aws_config::from_env().region(Region::new(self.region.clone()));
-            if let Some(credentials) = &self.credentials {
-                loader = loader.credentials_provider(SharedCredentialsProvider::new(
-                    aws_types::Credentials::from_keys(
-                        &credentials.access_key_id,
-                        &credentials.secret_access_key,
-                        credentials.session_token.clone(),
-                    ),
-                ));
-            }
+            let mut loader: aws_config::ConfigLoader = aws_config::from_env();
+            // RegionProviderChain is not Clone so we must construct a new on for each use
+            let region = || {
+                if let Some(region) = &self.region {
+                    RegionProviderChain::first_try(Region::new(region.clone()))
+                        .or_default_provider()
+                } else {
+                    let mut wc =
+                        aws_config::default_provider::region::DefaultRegionChain::builder();
+                    if let AwsCredentials::Profile { profile_name } = &self.credentials {
+                        wc = wc.profile_name(profile_name);
+                    }
+                    // TODO: this doesn't work for profiles which do not provide their own region
```

Comment:
Is this a bug? It seems like something we should file upstream.

Reply:
yeah the todo is to remind me to create a reproducible test case to see if this happens with plain boto.